flow/Platform.cpp

```diff
@@ -250,6 +283,29 @@ uint64_t getResidentMemoryUsage() {
 
 	rssize *= sysconf(_SC_PAGESIZE);
 
+	return rssize;
+#elif defined(__FreeBSD__)
+	uint64_t rssize = 0;
```

Comment:
Have you double checked that all of these functions actually return or calculate accurate information?  In looking at the various fields in "ProcessMetrics" and "MachineMetrics", I'm seeing some things that look odd.  "CPUSeconds" is "-2429.36".  Most of the disk-related fields are always 0 (e.g. DiskReads).  "CommittedMemory" says "7281307648", which optimistically assuming that's bytes, would be 67GB on my 8GB RAM VM.

Reply:
I'm not considering these metrics being wrong to be merge-blocking, but it's good to keep track of all the other pieces that need to get done for FreeBSD to be at feature parity with Linux.

Reply:
I did see that last night while testing. In this PR of course is a write-up of all the reasons why every routine was chosen as well as a single file where all the stats are sandbox and run in a standalone program. This should make it easy to figure out. The metric for RAM is a floating point miscalculation so it should have read 7.28GB I'll have to look at it. I distinctly remember this though where I thought there was a problem and moved the decimal so that is probably a typo or mistake somewhere. The CPUSeconds I know are being calculated with an extremely narrow boundary of time so I can pull that out a bit. I'll follow up with some more details on this later but by doing the writeup I actually have a good idea where these two metrics can be tightened up.

Reply:
I just caught an edit in getMachineRAMInfo which never got checked in page size was not being divided by 1024. I've updated the workbench [stats file](https://gist.github.com/wolfspider/3bd9fc54778d31b63ede362ddc77728e) I created here with the fix also..(starts at comment "RAM stats are off lets take a look").

Reply:
Memory is looking better this is verified. CPU seconds have changed but not fixed so I know which variable is causing it which has been changed to this in reference to the included screen:

    (uint64_t)(ms_per_transaction / 1000)

I've updated with the same routine from atop to compare better with Linux so the I/O per ms reading is off and this is the cause. In the linux code this is referred to as "Ticks" read from /proc/diskstats

Running the I/O per MS code in a standalone program in the terminal always comes back with 0 and will keep looking for an answer for that one.

<img width="1502" alt="memfix" src="https://user-images.githubusercontent.com/4861700/43361829-8f243122-92a9-11e8-9ee1-614a55a2b0c3.png">

Reply:
It seems that negative CPUSeconds are being affected by an equation based on CPU Totals. I'll post this as a question in the forums and reference it back here. The io/ms calculation is not affecting this after all but is a separate issue altogether.

Reply:
CPU totals look good and RAM totals look good now. The large number is due to FoundationDB needing bytes actually and can see the testing done to resolve this here:
https://forums.foundationdb.org/t/memory-available-in-fdbcli/653

CPU totals are being aggregated across all the cores successfully with the original code that's in there I've tested that recently by checking cp_times for individual cores and total cores:

    jessebennett@ptolemaic ~ $ sysctl kern.cp_times
    kern.cp_times: 41 0 722 819 625212 26 0 1052 75 625641
    jessebennett@ptolemaic ~ $ sysctl kern.cp_time
    kern.cp_time: 67 0 1778 895 1251595

Code was fixed recently there was something wrong with the equation and now CPUSeconds are reading positive again referenced here:

https://forums.foundationdb.org/t/getmachineload-and-calculation-of-cpuseconds-for-machinemetrics/603